### ISSUE TITLE
[2.0.6] Relative URL Redirection Broken When Specified with Default Slash

### ISSUE THREAD

**Issue by virgofx** [phalcon/cphalcon#10739]
See Issue #10610

Both functionality should behave the same for a base URI with nothing or a single '/'.

```php
// Set with leading slash - This used to be the default for all of Phalcon 1
$this->url->setBaseUri('/');

// Fixed in 10610:
$this->response->redirect('/test');
// Header: Location: /test

// Works
$this->response->redirect('');
// Header: Location: /

// Works
$this->response->redirect();
// Header: Location: /

//  Broken
$this->response->redirect('/');
// No response, phalcon dies somewhere
// Expected:  Header: Location: /
```

```php
// Set to empty
$this->url->setBaseUri('');

// Fixed in 10610:
$this->response->redirect('/test');
// Header: Location: /test

// Issue - Not sure?
$this->response->redirect('');
// Header: Location: ""
// Shouldn't the default redirect at least append a leading slash. Otherwise there is a redirect loop here in browser with empty location.

// Issue - Should behave same as above. In my opinion the location should be '/'
$this->response->redirect();
// Redirect loop - Same as above

//  Now Works since base URI is empty
$this->response->redirect('/');
```

**Comment by andresgutierrez** [comment]
> Both functionality should behave the same for a base URI with nothing or a single '/'.

Why?

**Comment by virgofx** [comment]
Because the response->redirect() ... should not take into account base URI.  

Even if it does .. there is a special case with `/`. The other issue fixed one case, there is just one more single case to handle.

For instance .... if base URI = `/`
$response->redirect('/test') === `/test`  (expected)  not `//test` (not expected) 

The major error is in the first block. When setting base URI to `/`. A redirect of `/` should also drop the double `//` just as it does with test; however, there is a core error somewhere as the page will not even get the response in this case.

**Comment by andresgutierrez** [comment]
> For instance .... if base URI = /
> $response->redirect('/test') === /test (expected) not //test (not expected)

This already works like that

**Comment by virgofx** [comment]
Yes, that was fixed in #10739. Just need to handle one extra case where it does not behave similarly

`$response->redirect('/')` --> Expected `Location: /`.    Actual : Phalcon dies, no response

**Comment by virgofx** [comment]
```
# Method                      # BaseURI   # Expected   # Actual   # Status 
$response->redirect('/test')              /test        /test      [good]
$response->redirect('/')                  /            /          [good]
$response->redirect('/test')  /           /test        /test      [good]
$response->redirect('/')      /           /            -          [broken]

```

**Comment by andresgutierrez** [comment]
Fixed in the 2.0.x branch

**Comment by virgofx** [comment]
@andresgutierrez Thank you
